src/features/sizes.ts

```diff
@@ -0,0 +1,80 @@
+import fs from 'node:fs/promises'
+import path from 'node:path'
+import { gzipSync } from 'node:zlib'
+import Debug from 'debug'
+import { formatBytes } from '../utils/format'
+import { logger } from '../utils/logger'
+
+const debug = Debug('tsdown:sizes')
+
+const DTS_EXTENSIONS = ['.d.ts', '.d.mts', '.d.cts']
+
+interface SizeInfo {
+  size: number
+  gzip: number
```

Comment:
I'd advocate for supporting `brotli` too!

Reply:
@TheAlexLichter I have just pushed changes to include brotli support as well